core/app/models/spree/tax/tax_helpers.rb

```diff
@@ -17,9 +17,9 @@ module TaxHelpers
       #
       # For further discussion, see https://github.com/spree/spree/issues/4397 and https://github.com/spree/spree/issues/4327.
       def applicable_rates(order)
-        order_zone_tax_categories = rates_for_order_zone(order).map(&:tax_category)
+        order_zone_tax_categories = rates_for_order_zone(order).flat_map(&:tax_categories)
```

Comment:
Yes I will add that as solidus 2.2 actually uses `rates_for_order(order).map(&:tax_category_id)`